pyomo/core/base/block.py

```diff
@@ -1334,10 +1414,15 @@ def _component_typemap(self, ctype=None, active=None, sort=False):
         else:
             return PseudoMap(self, ctype, active, sort)
 
-    def _component_data_iter(self, ctype=None, active=None, sort=False):
-        """
-        Generator that returns a 3-tuple of (component name, index value,
-        and _ComponentData) for every component data in the block.
+    def _component_data_iteritems(self, ctype, active, sort, dedup):
```

Comment:
Can you please add the expected types of the arguments (either as type hints or in the doc string)? I believe deduce is expected to be an instance of `_DeduplicateInfo`, but future me (if I ever look at this code again) will not remember that.